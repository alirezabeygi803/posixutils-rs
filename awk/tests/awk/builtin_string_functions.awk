BEGIN {
    print index("hello", "l");
    print length("hello");
    print match("hello", /l+/);
    print RSTART;
    print RLENGTH;
    split("hello test b", a);
    print a[1];
    print a[2];
    print a[3];
    split("hello,test,b", a, /,/);
    print a[1];
    print a[2];
    print a[3];
    print sprintf("format string");
    print sprintf("format string: %s", "hello");
    print sprintf("format string: %s %d", "hello", 1);
    print tolower("HELLO");
    print toupper("hello");
    s = "hello";
    gsub(/l/, "L", s);
    print s;
    s = "hello";
    sub(/l/, "L", s);
    print s;
    print substr("hello", 2, 2);
}

$1 == 1 { print length }
$1 == 2 { print length() }
$1 == 3 { gsub(/el|ra/, "xx"); print $0 }
$1 == 4 { sub(/a/, "xx"); print $0 }
