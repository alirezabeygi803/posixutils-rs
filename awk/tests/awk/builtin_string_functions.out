3
5
3
3
3
2
hello
test
b
hello
test
b
format string
format string: hello
format string: hello 1
hello
HELLO
heLLo
heLlo
el
17
17
3 Rosxxyn xxdiologist 28
4 Megxxn machinist 32
